torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2728,12 +2729,23 @@ def hint(s):
                 track_symint(source, t)
                 continue
             assert isinstance(t, torch.Tensor)
-            for i, ss in enumerate(t.size()):
-                property_source = TensorPropertySource(source, TensorProperty.SIZE, i)
-                track_symint(property_source, ss, constraint[i])
-            for i, ss in enumerate(t.stride()):
-                track_symint(TensorPropertySource(source, TensorProperty.STRIDE, i), ss)
-            track_symint(TensorPropertySource(source, TensorProperty.STORAGE_OFFSET), t.storage_offset())
+            if is_traceable_wrapper_subclass(t):
+                # If our placeholder is a tensor subclass, then the "true" symints
+                # come from the subclass's inner tensors.
+                source_to_inner_t, _ = t.__tensor_flatten__()
```

Comment:
Where is the logic that tests that the specs match?

Reply:
Yes I'm talking about this. Not only do you need to guard on subclassness, you need to guard on the spec being the same the next time around.